Document/0x06c-Reverse-Engineering-and-Tampering.md

```diff
@@ -50,6 +50,56 @@ Building a reverse engineering environment for free is possible. However, there
+
+![Ghidra import file](Images/Chapters/0x06c/Ghidra_import_binary.png)
+
+If the correct file is specified, Ghidra will show meta-information about the binary before starting the analysis. 
+
+![Ghidra Mach-O file import](Images/Chapters/0x06c/Ghidra_macho_import.png)
+
+To get the disassembled code for the binary file choosen above, double click the imported file. Click **yes** and **analyze** for auto-analysis on the subsequent windows. Auto-analysis will take some time depending on the size of the binary, the progress can be tracked in the bottom right corner of the code browser window. Once auto-analysis is completed you can start exploring the binary.  
+
+![Ghidra code browser window](Images/Chapters/0x06c/Ghidra_main_window.png)
+
+In above annotated main window of Ghidra, disassembly window and symbol tree window are most important for moving around and exploring the binary. Decompiler window shows the decompiled version of the function selected for disassembly. It is important to note decompilation is not always accurate, but nevertheless very helpful in getting a quick understanding of the function being analyzed. It is always a good idea to keep cross-checking decompiled code against disassembled code for maximum accuracy.
+
+There are many other functionalities available in Ghidra, you can explore them by checking *Window* in menu. For example, if you want to check the strings present in the binary, which can be very helpful in reversing a binary, use **Defined Strings** option.  
+
+![Ghidra strings window](Images/Chapters/0x06c/Ghidra_string_window.png)
+
+Often the core logic of the application is implemented in the frameworks and the app binary is just a wrapper around it. In such cases, the above approach can be used to analyze *framework* binaries as well. 
+
+##### Disassembling With IDA Pro
+If you have license for IDA Pro, you can analyze app binary using IDA Pro as well. 
+
+To get started with IDA Pro, simply open the app binary in IDA Pro. 
+
+![IDA Pro open a Mach-O file](Images/Chapters/0x06c/ida_macho_import.png)
+
+Upon opening the file, IDA will perform auto-analysis, this is a good time to take a coffee break and check your emails. Once analysis is completed you can browse the disassembly in the disassembly window and explore functions in function window, both shown in the screenshot below. 
+
+![IDA Pro main window](Images/Chapters/0x06c/ida_main_window.png)
+
+IDA Pro license does not include decompiler by default and requires an additional license for Hex-Rays decompiler, which is expensive as well. On the other hand, Ghidra comes with a very capable free inbuilt decompiler, making it a compelling alternative to use for reverse engineering.  
+
+If you have IDA Pro license and do not want to buy Hex-Rays decompiler, you can use Ghidra decompiler by installing [GhIDA plugin](https://github.com/Cisco-Talos/GhIDA/) for IDA. 
+
+> The freeware version of IDA Pro unfortunately does not support ARM processor type.
```

Comment:
Move this to the top.

Reply:
I have moved the "freeware" part to the top. Keeping the Ghida plugin at same place, as i think it is better to have after the discussion of hex-rays decompiler done in the previous paragraph.